src/Contract/Operation/Unpackable.php

```diff
@@ -12,14 +12,16 @@
 use loophp\collection\Contract\Collection;
 
 /**
- * @psalm-template TKey
- * @psalm-template TKey of array-key
- * @psalm-template T
+ * @template TKey of int
+ * @template T of array{0: NewTKey, 1: NewT}
+ *
+ * @template NewTKey of array-key
+ * @template NewT
```

Comment:
I literally must've spent over an hour getting these type hints right 😄

Reply:
but it's beautiful when the static analysis finally turns green ✅ 😆

Reply:
damn looks like this is not actually supported yet in PHPStan 😄. Not sure why it was showing all good in another file

Reply:
trying to work around it makes this worse and incorrect, so I think I'll keep this. Psalm supports it and it passes, and PHPStan should support it soon: https://github.com/phpstan/phpstan/issues/3931

Reply:
Can't wait for this to be supported !

Reply:
It's now supported! https://github.com/phpstan/phpstan-src/pull/673

Reply:
nice job! 💯